legend-engine-xt-relationalStore-grammar/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperRelationalBuilder.java

```diff
@@ -961,6 +966,20 @@ private static PropertyMapping processRelationalPropertyMapping(RelationalProper
             Assert.assertTrue(eMap != null, () -> "Can't find enumeration mapping '" + propertyMapping.enumMappingId + "'");
             res = res._transformer(eMap);
         }
+        try
```

Comment:
I am wondering if this whole logic should be enclosed in `if (context.pureModel.getWarnings().size() < PureModel.MAX_WARNINGS)`. `public static final String MAX_WARNINGS = 100` can be defined in PureModel. My reasoning is that, warnings beyond a threshold, don't give any extra nudge to the end users to fix their models. Also, this would make the check more efficient.